datatypes/src/collections/feature_collection.rs

```diff
@@ -458,7 +459,18 @@ where
                     arrow::compute::lt_utf8_scalar,
                 )?;
             }
-            FeatureDataType::Category => {
+            FeatureDataType::DateTime => {
+                apply_filters(
+                    as_primitive_array::<Date64Type>(column),
+                    &mut filter_array,
+                    ranges,
+                    arrow::compute::gt_eq_scalar,
+                    arrow::compute::gt_scalar,
+                    arrow::compute::lt_eq_scalar,
+                    arrow::compute::lt_scalar,
+                )?;
+            }
+            FeatureDataType::Category | FeatureDataType::Bool => {
```

Comment:
I tried using `arrow::array::cast::as_boolean_array` and then the comparisons. It does not compile because Rust requires that `FeatureDataValue `must be convertible to `BooleanArray`, which doesn't really makes sense to me.

Reply:
Currently, arrow provides no such scalar comparisons. gt_eq_bool only works with two arrays. For now I copied the changes from [this Pull Request](https://github.com/apache/arrow-rs/pull/977) and the `compare_op_scalar` macro (which is private). This needs to be fixed after an appropriate version of arrow gets released.
47a20c8